cwms_radar_api/src/main/java/cwms/radar/data/dao/LocationGroupDao.java

```diff
@@ -159,35 +171,46 @@ public List<LocationGroup> getLocationGroups(String officeId) {
-                        alga.GROUP_ID, alga.DB_OFFICE_ID, alga.LOCATION_ID, alga.ALIAS_ID,
+        SelectOnConditionStep<? extends Record> onStep = dsl.select(
+                        alcg.CAT_DB_OFFICE_ID,
+                        alcg.LOC_CATEGORY_ID,
+                        alcg.LOC_CATEGORY_DESC,
+                        alcg.GRP_DB_OFFICE_ID,
+                        alcg.LOC_GROUP_ID,
+                        alcg.LOC_GROUP_DESC,
+                        alcg.LOC_GROUP_ATTRIBUTE,
+                        alcg.SHARED_LOC_ALIAS_ID,
+                        alcg.SHARED_REF_LOCATION_ID,
+                        alga.DB_OFFICE_ID,
+                        alga.LOCATION_ID,
+                        alga.ALIAS_ID,
                         alga.ATTRIBUTE,
-                        alga.REF_LOCATION_ID, alga.SHARED_ALIAS_ID, alga.SHARED_REF_LOCATION_ID,
-                        alcg.CAT_DB_OFFICE_ID, alcg.GRP_DB_OFFICE_ID, alcg.LOC_CATEGORY_ID,
-                        alcg.LOC_CATEGORY_DESC, alcg.LOC_GROUP_DESC, alcg.LOC_GROUP_ATTRIBUTE)
+                        alga.REF_LOCATION_ID)
                 .from(alcg).leftJoin(alga)
-                .on(alcg.LOC_CATEGORY_ID.eq(alga.CATEGORY_ID)
-                        .and(alcg.LOC_GROUP_ID.eq(alga.GROUP_ID)));
+                            .on(alcg.LOC_CATEGORY_ID.eq(alga.CATEGORY_ID)
+                            .and(alcg.LOC_GROUP_ID.eq(alga.GROUP_ID)));
         if (officeId != null) {
             if (CWMS.equalsIgnoreCase(officeId)) {
                 connectBy = onStep.where(alcg.CAT_DB_OFFICE_ID.eq(CWMS)
                         .and(alcg.GRP_DB_OFFICE_ID.eq(CWMS)));
             } else {
                 connectBy = onStep.where(alcg.CAT_DB_OFFICE_ID.in(CWMS, officeId)
                         .and(alcg.GRP_DB_OFFICE_ID.in(CWMS, officeId))
-                        .and(alga.DB_OFFICE_ID.eq(officeId)));
+                        .and(alga.DB_OFFICE_ID.isNull().or(alga.DB_OFFICE_ID.eq(officeId))));
             }
         } else {
-            connectBy = onStep;
+            connectBy = onStep.where(alcg.LOC_GROUP_ID.isNotNull());
         }
 
-        connectBy.orderBy(alga.CATEGORY_ID, alga.GROUP_ID, alga.ATTRIBUTE).fetchSize(1000)    //
+        connectBy.orderBy(alcg.LOC_CATEGORY_ID, alcg.LOC_GROUP_ID, alga.ATTRIBUTE).fetchSize(1000)    //
                 // This made the query go from 2 minutes to 10 seconds?
                 .stream().map(mapper::map).forEach(pair -> {
                     LocationGroup locationGroup = pair.component1();
-                    List<AssignedLocation> list = map.computeIfAbsent(locationGroup,
-                            k -> new ArrayList<>());
-                    list.add(pair.component2());
+                    List<AssignedLocation> list = map.computeIfAbsent(locationGroup, k -> new ArrayList<>());
+                    AssignedLocation assignedLocation = pair.component2();
```

Comment:
Probably don't even make the new list if the assignedLocation is null.

Reply:
I still need the key in the map and don't want to insert a null collection.

Reply:
ok